### DISCUSSION TITLE
Access static JSON routes in build?

### DISCUSSION THREAD

**Discussion by ekafyi** [vercel/next.js#31366] [Help]
TL;DR: Given an SSG (getStaticProps) Nextjs web app, is it possible to access the generated JSON route from the client side?

Example route:
- `pages/posts/[slug].tsx`
- generated HTML route = `/posts/foo`
- generated JSON route = `/_next/data/development/posts/foo.json` (dev) or `/_next/data/Yh3PFIyMSOyeXCiOM6WUW/posts/foo.json` (build) <--- is it possible to access this from the client side, eg. with axios or swr library
   - _(Tried a simple fetch call before posting here, no luck, local relative urls are not allowed)_

Use case: I'd like to show a "preview" hover card (think Wikipedia article) on links in the post content. The post content is generated from Markdown; post A may contain links to posts B, C, D, and so on.

In my case, the data source is a local GraphQL API from a local Postgres DB, which I dont know how to deploy 😬 (hence the SSG approach). But on a more contextual level, when all the data you need _already exist_ in the JSON files generated from getStaticProps, IMO it would be redundant to set up an API route and refetch the data.

Or is there a different, better approach to handle this?

Related/similar question: https://github.com/vercel/next.js/discussions/26346

**Comment by ekafyi** [comment]
Quick idea, have not tried it: Would it be possible to generate custom build id https://nextjs.org/docs/api-reference/next.config.js/configuring-the-build-id and serve the posts paths as API routes? 🤔
